### ISSUE TITLE
About the reproduction of the experiment

### ISSUE THREAD

**Issue by hj-work** [WZH0120/SAM2-UNet#55]
Hello, I uncommented the random seed part in train.py and used single-GPU for training. However, due to certain factors—for example, the setting `dataloader = DataLoader(dataset, batch_size=args.batch_size, shuffle=True, num_workers=8)`—when I repeat the experiment, the training loss changes differently, and the prediction results also vary. I trained and tested the model on the Kvasir-SEG dataset for the polyp segmentation task, and the best result achieved is mDice: 0.925, which is slightly lower than the 0.928 reported in the paper.

I wonder, for the sake of fairness, should I train multiple times and take the average, or directly select the best-performing one? Is it acceptable to only train and test once if it's cumbersome? I'm new to this field, so my questions might be basic. I hope you don't mind.

Hello,

I have a question regarding experimental reproducibility. After uncommenting the random seed section in train.py and performing single-GPU training, I observed inconsistent training loss trajectories and varying model performance, potentially due to factors like the DataLoader configuration:
`dataloader = DataLoader(dataset, batch_size=args.batch_size, shuffle=True, num_workers=8)`

I trained and tested the model on the Kvasir-SEG dataset for polyp segmentation, the highest test performance achieved was mDice: 0.925, slightly lower than the 0.928 reported in the paper.

To ensure fair comparison:
Should I conduct multiple training runs and report the average performance?
Alternatively, is selecting the best-performing run acceptable?
Would performing only a single training/evaluation run be considered valid?

Apologies if this is too basic – I'm new to this area. Appreciate your guidance!

**Comment by xiongxyowo** [comment]
Q1: Should I conduct multiple training runs and report the average performance?
A1: For many existing papers, the results are reported based on the best run. If you explicitly report the average result (with mean±std), then the compared methods should also be retrained and report accordingly, which will be a large burden.

Q2: Alternatively, is selecting the best-performing run acceptable?
A2: I guess it's an ~~unspoken rule~~.

Q3: Would performing only a single training/evaluation run be considered valid?
A3: Sure.

Also, you are welcomed to join our [WeChat Discussion Group](https://github.com/WZH0120/SAM2-UNet/blob/main/wechat.jpg) :)
